acceptance-tests/drain_test.go

```diff
@@ -72,4 +73,52 @@ var _ = Describe("Drain Test", func() {
 		_, err = http.Get(fmt.Sprintf("http://%s", haproxyInfo.PublicIP))
 		expectConnectionRefusedErr(err)
 	})
+
+	// drain with an inexisting Process
+	FIt("Honors grace and drain periods with stale PID file", func() {
+		haproxyBackendPort := 12000
+		// Expect initial deployment to be failing due to lack of healthy backends
+		haproxyInfo, _ := deployHAProxy(baseManifestVars{
+			haproxyBackendPort:    haproxyBackendPort,
+			haproxyBackendServers: []string{"127.0.0.1"},
+			deploymentName:        deploymentNameForTestNode(),
+		}, []string{opsfileDrainTimeout}, map[string]interface{}{}, false)
+
+		// Verify that is in a failing state
+		Expect(boshInstances(deploymentNameForTestNode())[0].ProcessState).To(Or(Equal("failing"), Equal("unresponsive agent")))
+
+		closeLocalServer, localPort := startDefaultTestServer()
+		defer closeLocalServer()
+
+		closeTunnel := setupTunnelFromHaproxyToTestServer(haproxyInfo, haproxyBackendPort, localPort)
+		defer closeTunnel()
+
+		By("Waiting monit to report HAProxy is now healthy (due to having a healthy backend instance)")
+		// Since the backend is now listening, HAProxy healthcheck should start returning healthy
+		// and monit should in turn start reporting a healthy process
+		// We will up to wait one minute for the status to stabilise
+		Eventually(func() string {
+			return boshInstances(deploymentNameForTestNode())[0].ProcessState
+		}, time.Minute, time.Second).Should(Equal("running"))
+
+		By("The healthcheck health endpoint should report a 200 status code")
+		expect200(http.Get(fmt.Sprintf("http://%s:8080/health", haproxyInfo.PublicIP)))
+
+		By("Sending a request to HAProxy works")
+		expectTestServer200(http.Get(fmt.Sprintf("http://%s", haproxyInfo.PublicIP)))
+
+		// Stop HAproxy
+		_, _, err := runOnRemote(haproxyInfo.SSHUser, haproxyInfo.PublicIP, haproxyInfo.SSHPrivateKey, "sudo monit stop haproxy")
```

Comment:
```suggestion
		// Stop HAproxy
		_, _, err := runOnRemote(haproxyInfo.SSHUser, haproxyInfo.PublicIP, haproxyInfo.SSHPrivateKey, "monit stop haproxy")
		Expect(err).NotTo(HaveOccurred())
		
```

I would suggest checking for the `err` here just in case.

If that still doesn't work, output the 1st and 2nd returned values that contain stdout and stderr outputs of the SSH call.

`runOnRemote` is running as root, if I understand correctly, so no need for `sudo`.

Reply:
`runOnRemote` uses `vcap` user not `root`.